src/Query/Builder.php

```diff
@@ -279,6 +284,59 @@ public function dump(mixed ...$args)
         return $this;
     }
 
+    private function getAggregationBuilder(): AggregationBuilder
+    {
+        if (! trait_exists(FluentFactoryTrait::class)) {
+            throw new BadMethodCallException('Aggregation builder requires package mongodb/builder 0.2+');
+        }
+
+        $agg = new AggregationBuilder($this->collection, [], $this->options);
+
+        $wheres = $this->compileWheres();
+
+        if (count($wheres)) {
+            $agg->match(...$wheres);
+        }
+
+        // Distinct query
+        if ($this->distinct) {
+            // Return distinct results directly
+            $column = $this->columns[0] ?? '_id';
+
+            $agg->group(
+                _id: \MongoDB\Builder\Expression::fieldPath($column),
+                _document: Accumulator::first(Variable::root()),
+            );
+            $agg->replaceRoot(
+                newRoot: new FieldPath('_document'),
+            );
+        }
+
+        if ($this->orders) {
+            $agg->sort(...$this->orders);
+        }
+
+        if ($this->offset) {
+            $agg->skip($this->offset);
+        }
+
+        if ($this->limit) {
+            $agg->limit($this->limit);
+        }
+
+        // Normal query
+        // Add custom projections.
+        if ($this->projections || $this->columns) {
+            $columns = in_array('*', $this->columns) ? [] : $this->columns;
```

Comment:
What does an asterisk in `$this->columns` indicate? And is it valid for the asterisk to appear alongside other values in the array, or would it make sense to assert that it's the only element if present?

I'm curious as to why an asterisk results in _only_ `$this->projections` being used, instead of selecting _all_ fields (i.e. no projection at all).

Reply:
It has the SQL meaning: select all columns. In SQL if you write `SELECT *, foo` you get the `foo` column twice in the results.

In Eloquent (SQL or MongoDB), an empty list of columns means a select of all fields.